packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -241,21 +254,64 @@ export function createInsightsMiddleware<
           setUserTokenToSearch(anonymousUserToken, true);
         }
 
-        // We consider the `userToken` coming from a `init` call to have a higher
-        // importance than the one coming from the queue.
-        if (userTokenBeforeInit) {
-          setUserTokenToSearch(userTokenBeforeInit, true);
-          insightsClient('setUserToken', userTokenBeforeInit);
-        } else if (queuedUserToken) {
-          setUserTokenToSearch(queuedUserToken, true);
-          insightsClient('setUserToken', queuedUserToken);
+        function setUserToken(
+          token: string | number,
+          userToken?: string | number,
+          authenticatedUserToken?: string | number
```

Comment:
you mentioned the refactor, but I'm not sure if it's a good idea to have two adjacent optional params. Shouldn't the first userToken be required (but allowed to be undefined)?

Reply:
Isn't it the same? If not passed, the `userToken ` will be `undefined`.

Reply:
Yes, but there's no real use case for passing `authenticatedUserToken` if `userToken` isn't passed (you can't do it). If the signature is required (maybe even for both) we won't have weird edge cases if ever we call setUserToken without parameters

Reply:
Yep I think I see what you mean. This function is meant to be internal (and temporary) just to avoid repetition and further inflating bundle size. I can rewrite it more cleanly later.